distribution/push.go

```diff
@@ -127,6 +133,19 @@ func Push(ctx context.Context, ref reference.Named, imagePushConfig *ImagePushCo
 			continue
 		}
 
+		parsedURL, urlErr := url.Parse(endpoint.URL)
```

Comment:
@calavera: I agree that it would make more sense for `APIEndpoint` to contain a parsed URL. I added a commit that makes this change.

Please let me know whether:

a) the change looks good
b) this should be broken out into a separate PR